src/rust/src/lib.rs

```diff
@@ -16,46 +16,29 @@ fn hello_world() -> &'static str {
 /// @export
 #[extendr]
 pub fn csv_to_geojson() {
-    // // Test data:
-    // let geometry = Geometry::new(Value::Point(vec![-120.66029, 35.2812]));
-    // let geojson = GeoJson::Feature(Feature {
-    //     bbox: None,
-    //     geometry: Some(geometry),
-    //     id: None,
-    //     properties: None,
-    //     foreign_members: None,
-    // });
-    // let geojson_string = geojson.to_string();
-    // serde_json::to_writer_pretty(&mut File::create("points_rust.geojson").unwrap(), &geojson_string).unwrap();
-
-    // let mut points = Vec::new();
-    let mut points_geojson = Vec::new();
     let mut reader = csv::Reader::from_path("points.csv").unwrap();
-    
-  for result in reader.records() {
-        let record = result.unwrap();
-        let lat = record[0].parse::<f64>().unwrap();
-        let lon = record[1].parse::<f64>().unwrap();
-        // points.push(Point::new(lon, lat));
-        points_geojson.push(Geometry::new(Value::Point(vec![lon, lat])));
-
-    }
-    let geojson = points_geojson.into_iter().collect::<GeoJson>();
-    // print!("{}", to_string_pretty(&geojson).unwrap());
-    // let geojson_string = to_string_pretty(&geojson).unwrap();
-    serde_json::to_writer_pretty(&mut File::create("points_rust.geojson").unwrap(), &geojson).unwrap();
-    // let geojson = GeoJson::Feature(Feature {
-    //     bbox: None,
-    //     geometry: Some(Geometry::new(Value::MultiPoint(points_value))),
-    //     id: None,
-    //     properties: None,
-    //     foreign_members: None,
-    // });
-    // let gjstring = to_string_pretty(&geojson).unwrap();
-    // println!("{}", gjstring);
-    // serde_json::to_writer_pretty(&File::create("points_rust.geojson"), &geojson);
-    // geojson.to_file("points_rust.geojson").unwrap();
-    // geojson_string.to_file("points_rust.geojson");
+    let points = reader
+        .records()
+        // this will silently discard invalid / unparseable records
+        .filter_map(|record| record.ok())
+        .map(|record| {
+            Feature::from(Value::from(Point::new(
+                record[1].parse::<f64>().unwrap(),
+                record[0].parse::<f64>().unwrap(),
+            )))
+        })
+        .collect();
```

Comment:
This syntax is really nice, kind of pipe with the dot syntax. I need to learn this!